src/Microsoft.AspNetCore.Mvc.Core/Internal/FilterActionInvoker.cs

```diff
@@ -336,18 +337,6 @@ private Task InvokeAllResourceFiltersAsync()
                 }
                 else
                 {
-                    // We've reached the end of resource filters, so move to setting up state to invoke model
-                    // binding.
-                    var valueProviders = new List<IValueProvider>();
-                    var factoryContext = new ValueProviderFactoryContext(Context);
-
-                    for (var i = 0; i < _valueProviderFactories.Count; i++)
-                    {
-                        var factory = _valueProviderFactories[i];
-                        await factory.CreateValueProviderAsync(factoryContext);
-                    }
-                    Context.ValueProviders = factoryContext.ValueProviders;
```

Comment:
Previously the value providers were not created until this point, immediately before model binding. In the normal (non-`TryUpdateModelAsync()`) case, what is better with this delayed until w/in model binding?